pkg/policy/repository.go

```diff
@@ -618,6 +621,8 @@ func (p *Repository) ResolvePolicy(id uint16, labels labels.LabelArray, policyOw
 					TrafficDirection: trafficdirection.Ingress.Uint8(),
 				}
 				calculatedPolicy.PolicyMapState[keyToAdd] = MapStateEntry{}
+			} else if ingressAccess == api.Denied {
+				calculatedPolicy.DeniedIngressIdentities[identity] = labels
```

Comment:
Don't you have to worry about `api.Undecided` return values here? If not, then I'd change this to an `else` without the `if` condition and add a comment that explains why it is valid. Otherwise add a comment after that explains what we do about `api.Undecided` values (nothing?).

Reply:
We should explicitly only add an identity to the denied set if the `api.Denied` verdict is returned. `api.Undecided` is only returned when a specific decision hasn't been returned yet. 

I will add the comment in an upcoming commit.